src/base/default_coupling.C

```diff
@@ -93,8 +100,11 @@ void DefaultCoupling::operator()
   if (!this->_n_levels)
     {
       for (const auto & elem : as_range(range_begin, range_end))
+      {
+        //libmesh_assert(_mesh->query_elem_ptr(elem->id()) ==elem);
```

Comment:
And same here? Right now MOOSE would actually fail this assertion because of our badness with displaced mesh?

Reply:
> So we will be able to uncomment this eventually though right?

Yes, when the MOOSE PR is merged https://github.com/idaholab/moose/pull/15655

Reply:
The displaced mesh and the reference mesh need to be synced correctly in a geometric sense. `Default_coupling` seems to be fine even when they are inconsistent, but it is still a good idea to make then consistent.